exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/EndToEndTest.java

```diff
@@ -54,50 +54,37 @@
   private static final long START_EPOCH_NANOS = TimeUnit.SECONDS.toNanos(3000) + 200;
   private static final long END_EPOCH_NANOS = TimeUnit.SECONDS.toNanos(3001) + 255;
   private static final StatusData SPAN_DATA_STATUS = StatusData.ok();
-  private static final String LOCALHOST = "127.0.0.1";
 
   private MockCloudTraceClient mockCloudTraceClient;
   private TraceExporter exporter;
-  private Process mockServerProcess;
 
-  @Before
-  public void setup() throws MockServerStartupException {
-    try {
-      // Find a free port to spin up our server at.
-      ServerSocket socket = new ServerSocket(0);
-      int port = socket.getLocalPort();
-      String address = String.format("%s:%d", LOCALHOST, port);
-      socket.close();
-
-      // Start the mock server. This assumes the binary is present and in $PATH.
-      // Typically, the CI will be the one that curls the binary and adds it to $PATH.
-      String[] cmdArray =
-          new String[] {System.getProperty("mock.server.path"), "-address", address};
-      ProcessBuilder pb = new ProcessBuilder(cmdArray);
-      pb.redirectErrorStream(true);
-      mockServerProcess = pb.start();
-
-      // Setup the mock trace client.
-      mockCloudTraceClient = new MockCloudTraceClient(LOCALHOST, port);
+  /** A test-container instance that loads the Cloud-Ops-Mock server container. */
+  private static class CloudOperationsMockContainer
```

Comment:
Is it a lot of work to re-use this test setup in both tests?

Reply:
Right now, yes.  I'll do that in a follow on Cl as I have to restructure the entire build for non-published components.  I had tried to do it but ran into gradle hell and put that off.

I agree it should be done, not in this PR.